pkg/apis/intevents/v1alpha1/brokercell_types.go

```diff
@@ -70,8 +70,8 @@ var (
 
 // SystemResource specifies quantities for resources such as CPU and memory
 type SystemResource struct {
-	CPU    *string `json:"cpu,omitempty"`
-	Memory *string `json:"memory,omitempty"`
+	CPU    string `json:"cpu,omitempty"`
+	Memory string `json:"memory,omitempty"`
 }
```

Comment:
We previously made this because it allows us to set CPU/memory defaults separately. Now since we either set defaults for the whole component or not, maybe we should make it flat again? Like putting every parameter in the `ComponentParameters`.

Reply:
I think in the previously suggested structure there is additional value: the nested structure mimics the [Kubernetes format for defining requests and limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits), which I suspect the target audience for the feature might be more comfortable with, compared to our custom structure. No problem with reverting this though. What do you think?

Reply:
Sorry I don't recall this purpose. After all this will be a very different API (than the k8s API). The current form is also not similar to the k8s API. Not even to mention the HPA part. So I wouldn't worry about difference being a problem.

Reply:
I meant when it comes to requests and limits for CPU and memory, the HPA format is different indeed. Ok, sounds like we'd rather switch to the flat structure, I'll follow up on this.

Reply:
Done.